cmd/podman/shared/funcs.go

```diff
@@ -6,20 +6,22 @@ import (
 	"path/filepath"
 	"strings"
 
+	"github.com/containers/libpod/pkg/util"
 	"github.com/google/shlex"
 )
 
 func GetAuthFile(authfile string) string {
 	if authfile != "" {
 		return authfile
 	}
+
 	authfile = os.Getenv("REGISTRY_AUTH_FILE")
 	if authfile != "" {
 		return authfile
 	}
-	runtimeDir := os.Getenv("XDG_RUNTIME_DIR")
-	if runtimeDir != "" {
-		return filepath.Join(runtimeDir, "containers/auth.json")
+
+	if runtimeDir, err := util.GetRootlessRuntimeDir(); err == nil {
```

Comment:
The comment to that function says
```go
// GetRootlessRuntimeDir returns the runtime directory when running as non root
```
and callers in `libpod/runtime.go` are all guarded by `rootless.IsRootless()`.  OTOH `libpod/oci*.go` seems to call it without that guard.

I don’t know which one is right, but clarifying that one way or the other would be nice.

Reply:
We should probably make this the GetRuntimeDir function.  It seems to return the right thing.  I tried this out in a bunch of different ways including as root.

Reply:
@giuseppe WDYT?

Reply:
yes, I think we can rename it to `GetRuntimeDir()` given it works also as root